sstables/generation_type.hh

```diff
@@ -51,6 +51,14 @@ public:
         }
         return _value.get_least_significant_bits();
     }
+    /// tell if a generation identifier is supposed to be handled by the current shard
+    ///
+    /// the integer based generation is used as a hint to tell if the SSTable with
+    /// this generation should be handled by a given shard. but this does not applies
+    /// UUID based generation.
+    bool maybe_owned_by_this_shard() const {
+        return as_int() % smp::count == seastar::this_shard_id();
```

Comment:
IMO, this belongs elsewhere (sstable_directory?). With object storage and/or tablets, we'll use a different method.  generation_type doesn't know anything about how we distribute sstables across shards.

Reply:
Currently, the knowledge of assigning a generation to an sstable belonging to a certain shard is encoded in generation generator, which is a class closely associated with `generation_type`.

Reply:
We could move it there.

Reply:
sure. moved.